vm/devices/user_driver/src/lib.rs

```diff
@@ -35,7 +35,8 @@ pub trait DeviceBacking: 'static + Send + Inspect {
     /// Maps a BAR.
     fn map_bar(&mut self, n: u8) -> anyhow::Result<Self::Registers>;
 
-    fn get_dma_client(&self) -> Option<Arc<dyn DmaClient>>;
+    /// DMA Client for the device.
+    fn dma_client(&self) -> anyhow::Result<Arc<dyn DmaClient>>;
```

Comment:
Again, this method should not be failable. It should just return `Arc<dyn DmaClient>`.

Reply:
Better yet, I would rather have it return `DmaClient` and have the associated type, like the code did before. Why did you make it an Arc? Let the caller decide if it needs to put it behind an Arc (but most probably will not).

Reply:
talked offline - confusingly VfioDevice itself uses dyn dispatch to allocate/restore dma buffers. I wonder if we want to remove the vtable dispatch here in the future.